### PR TITLE
Clean Permission domain codebase

### PR DESCRIPTION
**Pull request by teryk**
- PermissionTestSuite successfully launched

### CODE REVIEW

**Comment by teryk** [comment]
@simonbrandhof I didn't have time to rename the column in permission_template table, kee->uuid but I deprecated the method in the dto

**Comment by teryk** [comment]
@stas-vilchik could you please check why the navigation relies on RoR for the global permissions and project permissions pages ?

File: sonar-db/src/main/java/org/sonar/db/permission/PermissionTemplateDto.java [hunk 1]
```diff
@@ -62,27 +55,35 @@ public PermissionTemplateDto setName(String name) {
...
   /**
-   * @since 5.2 the kee column is a proper uuid. Before that it was build on the name + timestamp
+   * @deprecated since 5.2 use {@link #setUuid(String)}
    */
-  public String getUuid() {
-    return kee;
-  }
-
+  @Deprecated
   public PermissionTemplateDto setKee(String kee) {
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/permission/PermissionTemplateDto.java]: INFO Do not forget to remove this deprecated code someday. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

File: sonar-db/src/main/java/org/sonar/db/permission/PermissionTemplateDto.java [hunk 2]
```diff
@@ -62,27 +55,35 @@ public PermissionTemplateDto setName(String name) {
     return this;
   }
 
+  /**
+   * @deprecated since 5.2 use {@link #getUuid()}
+   */
+  @Deprecated
   public String getKee() {
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/permission/PermissionTemplateDto.java]: INFO Do not forget to remove this deprecated code someday. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues:
* MINOR 1 minor
* INFO 2 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Immediately return this expression instead of assigning it to the temporary variable "effectiveTemplateUuid".](https://github.com/SonarSource/sonarqube/blob/783e75c3c8c3660e938597c372bd7f5e13cc777d/server/sonar-server/src/main/java/org/sonar/server/permission/ws/template/DefaultPermissionTemplateFinder.java#L87) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1488)

**Comment by simonbrandhof** [comment]
@teryk why deprecating the DTO method as it's internal ? Just replace the callers and delete the method.

**Comment by teryk** [comment]
nope, the column is still kee and not uuid

**Comment by teryk** [comment]
this is the best I could do with the time allowed

**Comment by simonbrandhof** [comment]
I think mybatis does not use methods but fields, but ok, forget.

**Comment by teryk** [comment]
well I remember doing it and you thought it was misleading so I left the getter and setter. MyBatis use the field if it does not find any getter/setter

**Comment by simonbrandhof** [comment]
- PermissionServer#globalPermissions() must be dropped. It requires to inline the constant ```GlobalPermissions.ALL``` in need_authorization.rb
- Functions ```permissions``` and ```permission_templates``` must be removed from internal.rb 
- PermissionService and PermissionUpdater should be merged as difference is not intuitive.
- Parameter ```updateProjectAuthorizationDate``` is not used in ```PermissionRepository#insertGroupPermission()```

**Comment by teryk** [comment]
@stas-vilchik could you please check why the navigation relies on RoR for the global permissions and project permissions pages ?

File: server/sonar-server/src/main/java/org/sonar/server/permission/ws/GroupsAction.java [hunk 3]
```diff
@@ -97,25 +99,22 @@ public void handle(Request wsRequest, Response wsResponse) throws Exception {
     }
   }
 
-  private WsGroupsResponse buildResponse(PermissionQuery permissionQuery, PermissionRequest permissionRequest) {
-    GroupWithPermissionQueryResult groupsResult = permissionFinder.findGroupsWithPermission(permissionQuery);
-    List<GroupWithPermission> groupsWithPermission = groupsResult.groups();
-
+  private WsGroupsResponse buildResponse(List<GroupWithPermissionDto> groupsWithPermission, PermissionRequest permissionRequest, int total) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/permission/ws/GroupsAction.java]: MAJOR Make "buildResponse" a "static" method. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2325)

File: server/sonar-server/src/main/java/org/sonar/server/permission/ws/UsersAction.java [hunk 4]
```diff
@@ -97,28 +100,25 @@ public void handle(Request wsRequest, Response wsResponse) throws Exception {
     }
   }
 
-  private WsUsersResponse wsUsersResponse(PermissionQuery permissionQuery, int page, int pageSize) {
-    UserWithPermissionQueryResult usersResult = permissionFinder.findUsersWithPermission(permissionQuery);
-    List<UserWithPermission> usersWithPermission = usersResult.users();
-
+  private WsUsersResponse buildResponse(List<UserWithPermissionDto> usersWithPermission, Paging paging) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/permission/ws/UsersAction.java]: MAJOR Make "buildResponse" a "static" method. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 10 issues:
* MAJOR 7 major
* MINOR 1 minor
* INFO 2 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'src/main/gen-java/org/sonar/batch/protocol' package (org.codehaus.sonar:sonar-batch-protocol:src/main/gen-java/org/sonar/batch/protocol) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'src/main/gen-java/org/sonar/batch/protocol/input' package (org.codehaus.sonar:sonar-batch-protocol:src/main/gen-java/org/sonar/batch/protocol/input) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'src/main/gen-java/org/sonar/batch/protocol/output' package (org.codehaus.sonar:sonar-batch-protocol:src/main/gen-java/org/sonar/batch/protocol/output) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'src/main/gen-java/org/sonar/db/protobuf' package (org.codehaus.sonar:sonar-db:src/main/gen-java/org/sonar/db/protobuf) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1228)
* MINOR [Immediately return this expression instead of assigning it to the temporary variable "effectiveTemplateUuid".](https://github.com/SonarSource/sonarqube/blob/b5c3c914780df64cfb5fa8e18ed8f3ad743a9b99/server/sonar-server/src/main/java/org/sonar/server/permission/ws/template/DefaultPermissionTemplateFinder.java#L87) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1488)
* MAJOR Add a 'package-info.java' file to document the 'src/main/gen-java/org/sonarqube/ws' package (org.codehaus.sonar:sonar-ws:src/main/gen-java/org/sonarqube/ws) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1228)

**Comment by SonarTech** [comment]
SonarQube analysis reported 8 issues:
* MAJOR 5 major
* MINOR 1 minor
* INFO 2 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'src/main/gen-java/org/sonar/batch/protocol' package (org.codehaus.sonar:sonar-batch-protocol:src/main/gen-java/org/sonar/batch/protocol) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'src/main/gen-java/org/sonar/batch/protocol/input' package (org.codehaus.sonar:sonar-batch-protocol:src/main/gen-java/org/sonar/batch/protocol/input) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'src/main/gen-java/org/sonar/batch/protocol/output' package (org.codehaus.sonar:sonar-batch-protocol:src/main/gen-java/org/sonar/batch/protocol/output) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'src/main/gen-java/org/sonar/db/protobuf' package (org.codehaus.sonar:sonar-db:src/main/gen-java/org/sonar/db/protobuf) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1228)
* MINOR [Immediately return this expression instead of assigning it to the temporary variable "effectiveTemplateUuid".](https://github.com/SonarSource/sonarqube/blob/983dc2b10b9f378db20032a2020a5c95f682c351/server/sonar-server/src/main/java/org/sonar/server/permission/ws/template/DefaultPermissionTemplateFinder.java#L87) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1488)
* MAJOR Add a 'package-info.java' file to document the 'src/main/gen-java/org/sonarqube/ws' package (org.codehaus.sonar:sonar-ws:src/main/gen-java/org/sonarqube/ws) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1228)

**Comment by simonbrandhof** [comment]
@teryk still not merged ?

**Comment by stas-vilchik** [comment]
@simonbrandhof @teryk the PR can't be merged, because it cleans up the page which is not rewritten yet (permission templates). The plan was to quickly rewrite that page, but then we realized that some WS were missing, so the page couldn't be finished.

File: server/sonar-server/src/main/java/org/sonar/server/permission/ws/TemplateUsersAction.java [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+      PermissionQuery query = buildQuery(dbSession, wsRequest, template);
+      WsTemplateUsersResponse templateUsersResponse = buildResponse(dbSession, query, template);
+      writeProtobuf(templateUsersResponse, wsRequest, wsResponse);
+    } finally {
+      dbClient.closeSession(dbSession);
+    }
+  }
+
+  private PermissionQuery buildQuery(DbSession dbSession, Request wsRequest, PermissionTemplateDto template) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/permission/ws/TemplateUsersAction.java]: MAJOR Make "buildQuery" a "static" method. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2325)

File: server/sonar-server/src/main/java/org/sonar/server/permission/ws/TemplateUsersAction.java [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+import static org.sonar.server.permission.ws.WsPermissionParameters.PARAM_PERMISSION;
+import static org.sonar.server.permission.ws.WsPermissionParameters.createProjectPermissionParameter;
+import static org.sonar.server.permission.ws.WsPermissionParameters.createTemplateParameters;
+import static org.sonar.server.ws.WsUtils.writeProtobuf;
+
+public class TemplateUsersAction implements PermissionsWsAction {
+
+  private final DbClient dbClient;
+  private final UserSession userSession;
+  private final PermissionFinder permissionFinder;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/permission/ws/TemplateUsersAction.java]: MAJOR Remove this unused "permissionFinder" private field. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 2 major
* MINOR 1 minor
* INFO 2 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Immediately return this expression instead of assigning it to the temporary variable "effectiveTemplateUuid".](https://github.com/SonarSource/sonarqube/blob/db4c66894f38bcbeb4b67420d96aa479d627d560/server/sonar-server/src/main/java/org/sonar/server/permission/ws/template/DefaultPermissionTemplateFinder.java#L87) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1488)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues:
* MINOR 1 minor
* INFO 2 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Immediately return this expression instead of assigning it to the temporary variable "effectiveTemplateUuid".](https://github.com/SonarSource/sonarqube/blob/967511afe96f08bf3d7ca1c6a734ebffea0e343f/server/sonar-server/src/main/java/org/sonar/server/permission/ws/template/DefaultPermissionTemplateFinder.java#L87) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1488)

File: server/sonar-web/src/main/js/apps/permission-templates/users-view.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+import './templates';
+
+export default Modal.extend({
+  template: Templates['permission-templates-users'],
+
+  onRender: function () {
+    this._super();
+    var searchUrl = baseUrl + '/api/permissions/template_users?ps=100&permission=' + this.options.permission +
+        '&templateId=' + this.options.permissionTemplate.id;
+    new window.SelectList({
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/permission-templates/users-view.js]: MAJOR Replace by a standard call to the function. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=javascript%3AConstructorFunctionsForSideEffects)

File: server/sonar-web/src/main/js/apps/permission-templates/groups-view.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+  return baseUrl + '/api/permissions/template_groups?ps=100&permission=' + permission +
+      '&templateId=' + permissionTemplate.id;
+}
+
+export default Modal.extend({
+  template: Templates['permission-templates-groups'],
+
+  onRender: function () {
+    this._super();
+    new window.SelectList({
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/permission-templates/groups-view.js]: MAJOR Replace by a standard call to the function. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=javascript%3AConstructorFunctionsForSideEffects)

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 2 major
* MINOR 1 minor
* INFO 2 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Immediately return this expression instead of assigning it to the temporary variable "effectiveTemplateUuid".](https://github.com/SonarSource/sonarqube/blob/de112666f61385364c5e8d9ed6ffa9a6106e2cfd/server/sonar-server/src/main/java/org/sonar/server/permission/ws/template/DefaultPermissionTemplateFinder.java#L87) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1488)

**Pull request merged**